test/blackbox/common/RTPSWithRegistrationWriter.hpp

```diff
@@ -400,6 +407,16 @@ class RTPSWithRegistrationWriter
         return *this;
     }
 
+    bool is_matched() const
```

Comment:
I think it might be useful in the future to have a `get_matched()` instead, and then make the comparison in the tests; same goes for the reader